python/cuml/metrics/_ranking.py

```diff
@@ -21,6 +21,83 @@
 import math
 
 
+@with_cupy_rmm
+def precision_recall_curve(y_true, probs_pred):
+    """Compute precision-recall pairs for different probability thresholds
+    Note: this implementation is restricted to the binary classification task.
+    The precision is the ratio ``tp / (tp + fp)`` where ``tp`` is the number of
+    true positives and ``fp`` the number of false positives. The precision is
+    intuitively the ability of the classifier not to label as positive a sample
+    that is negative.
+    The recall is the ratio ``tp / (tp + fn)`` where ``tp`` is the number of
+    true positives and ``fn`` the number of false negatives. The recall is
+    intuitively the ability of the classifier to find all the positive samples.
+    The last precision and recall values are 1. and 0. respectively and do not
+    have a corresponding threshold.  This ensures that the graph starts on the
+    y axis.
+    Read more in the :ref:`User Guide <precision_recall_f_measure_metrics>`.
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        True binary labels, {0, 1}.
+    probas_pred : array, shape = [n_samples]
+        Estimated probabilities or decision function.
+    Returns
+    -------
+    precision : array, shape = [n_thresholds + 1]
+        Precision values such that element i is the precision of
+        predictions with score >= thresholds[i] and the last element is 1.
+    recall : array, shape = [n_thresholds + 1]
+        Decreasing recall values such that element i is the recall of
+        predictions with score >= thresholds[i] and the last element is 0.
+    thresholds : array, shape = [n_thresholds <= len(np.unique(probas_pred))]
+        Increasing thresholds on the decision function used to compute
+        precision and recall.
+    --------
+    >>> import numpy as np
+    >>> from cuml.metrics import precision_recall_curve
+    >>> y_true = np.array([0, 0, 1, 1])
+    >>> y_scores = np.array([0.1, 0.4, 0.35, 0.8])
+    >>> precision, recall, thresholds = precision_recall_curve(
+    ...     y_true, y_scores)
+    >>> precision
+    array([0.66666667, 0.5       , 1.        , 1.        ])
+    >>> recall
+    array([1. , 0.5, 0.5, 0. ])
+    >>> thresholds
+    array([0.35, 0.4 , 0.8 ])
```

Comment:
check the format of examples in other classes to format correctly, for example https://github.com/rapidsai/cuml/blob/2b7f38a4b6e8de77564c75f1caf8d715268c9903/python/cuml/cluster/dbscan.pyx#L90